src/openms/include/OpenMS/FILTERING/CALIBRATION/InternalCalibration.h

```diff
@@ -70,402 +70,362 @@ namespace OpenMS
 
+     The calibration function is calculated for the whole map.
+     For the matching of the reference masses and the peaks the parameter mz_tolerance is used to
+     calculate a window around the reference masses. If more than one peak is found within this window the
+     closest peak is taken.
 
+     @param exp The peak map to calibrate
+     @param ref_masses The reference m/z values
+     @param trafo_file_name Output file to store the transformation function
+    */
     template <typename InputPeakType>
-    void calibrateMapList(std::vector<MSExperiment<InputPeakType> > & exp_list, std::vector<MSExperiment<InputPeakType> > & calibrated_exp_list, std::vector<double> & ref_masses, std::vector<double> & detected_background_masses);
-
-
-protected:
-
-    /// the actual calibration function
-    void makeLinearRegression_(std::vector<double> & observed_masses, std::vector<double> & theoretical_masses);
-
-    /// check if reference ids contain RT and MZ information as meta values
-    void checkReferenceIds_(std::vector<PeptideIdentification> & pep_ids);
-
-    /// check if reference ids contain RT and MZ information as meta values
-    void checkReferenceIds_(const FeatureMap & feature_map);
-
-    /// apply transformation to all features (including subordinates and convex hulls)
-    void applyTransformation_(const FeatureMap & feature_map, FeatureMap & calibrated_feature_map);
-
-    /// here the transformation is stored
-    TransformationDescription trafo_;
-  }; // class InternalCalibration
-
-
-  template <typename InputPeakType>
-  void InternalCalibration::calibrateMapSpectrumwise(const MSExperiment<InputPeakType> & exp, MSExperiment<InputPeakType> & calibrated_exp, std::vector<double> & ref_masses)
-  {
-#ifdef DEBUG_CALIBRATION
-    std::cout.precision(writtenDigits<double>(0.0));
-#endif
-    if (exp.empty())
-    {
-      std::cout << "Input is empty." << std::endl;
-      return;
-    }
-
-    if (exp[0].getType() != SpectrumSettings::PEAKS)
-    {
-      std::cout << "Attention: this function is assuming peak data." << std::endl;
-    }
-    calibrated_exp = exp;
-
-    Size num_ref_peaks = ref_masses.size();
-    bool use_ppm = (param_.getValue("mz_tolerance_unit") == "ppm") ? true : false;
-    double mz_tol = param_.getValue("mz_tolerance");
-    startProgress(0, exp.size(), "calibrate spectra");
-    // for each spectrum
-    for (Size spec = 0; spec <  exp.size(); ++spec)
+    void calibrateMapGlobally(MSExperiment<InputPeakType>& exp, 
+      const std::vector<double>& ref_masses,
+      const String& trafo_file_name = "")
     {
-      // calibrate only MS1 spectra
-      if (exp[spec].getMSLevel() != 1)
+      if (exp.empty())
       {
-        continue;
+        std::cout << "Input is empty." << std::endl;
+        return;
       }
 
+      if (exp[0].getType() != SpectrumSettings::PEAKS)
+      {
+        std::cout << "Attention: this function is assuming peak data." << std::endl;
```

Comment:
no, I think the estimation of spectrum type is known buggy and it should *not* fail simply because it assumes something